extensions/amp-analytics/0.1/test/test-linker-manager.js

```diff
@@ -177,6 +179,43 @@ describe('Linker Manager', () => {
     });
   });
 
+
+  it('should add linker with no defaultDomain and matching hostname', () => {
+    const config = {
+      linkers: {
+        testLinker1: {
+          enabled: true,
+          ids: {
+            _key: 'CLIENT_ID(_ga)',
+            gclid: '234',
+          },
+        },
+      },
+    };
+
+    const a = {
+      href: 'https://www.example.com',
+      hostname: 'www.example.com',
+    };
+
+    const manager = new LinkerManager(ampdoc, config);
+    sandbox.stub(manager, 'isLegacyOptIn_').returns(false);
+    const expandStub = sandbox.stub(manager, 'expandTemplateWithUrlParams_');
+    expandStub.withArgs('CLIENT_ID(_ga)')
+        .returns('amp-12345');
+    expandStub.returnsArg(0);
+    docInfoStub.returns({
+      sourceUrl: 'https://www.example.com/foo/bar',
+      canonicalUrl: 'https://www.example.com/foo/bar',
+    });
+    manager.init();
+
+    return Promise.all(manager.allLinkerPromises_).then(() => {
+      manager.handleAnchorMutation(a);
+      return expect(a.href).to.not.equal('https://www.example.com');
```

Comment:
compare to the full URL

Reply:
I was trying to avoid testing the linker in the linker-manager, but I agree it is probably better. Done.